Pod/Classes/PINAlternateRepresentationDelegate.h

```diff
@@ -0,0 +1,25 @@
+//
+//  PINAlternateRepresentationDelegate.h
+//  Pods
+//
+//  Created by Garrett Moon on 3/17/16.
+//
+//
+
+#import <Foundation/Foundation.h>
+
+#import "PINRemoteImageManager.h"
+
+@protocol PINRemoteImageManagerAlternateRepresentationDelegate <NSObject>
+@required
+
+/**
+ @discussion This method will be called with data off the wire or stored in the cache. Return an object to have it returned as the alternativeRepresentation object in the PINRemoteImageManagerResult. @warning this method can be called on the main thread, be careful of doing expensive work.
+ */
+- (id)alternateRepresentationWithData:(NSData *)data options:(PINRemoteImageManagerDownloadOptions)options;
```

Comment:
I don't think so. The returned object isn't stored in the disk cache, so it doesn't need to conform to NSCoding. Otherwise, the only thing we do with this object is pass it through as a result, so it should be able to be an arbitrary object.